src/coreComponents/constitutive/solid/DruckerPrager.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+ * ------------------------------------------------------------------------------------------------------------
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * Copyright (c) 2018-2019 Lawrence Livermore National Security LLC
+ * Copyright (c) 2018-2019 The Board of Trustees of the Leland Stanford Junior University
+ * Copyright (c) 2018-2019 Total, S.A
+ * Copyright (c) 2019-     GEOSX Contributors
+ * All right reserved
+ *
+ * See top level LICENSE, COPYRIGHT, CONTRIBUTORS, NOTICE, and ACKNOWLEDGEMENTS files for details.
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ *  @file DruckerPrager.cpp
+ */
+
+#include "DruckerPrager.hpp"
+
+namespace geosx
+{
+using namespace dataRepository;
+namespace constitutive
+{
+
+DruckerPrager::DruckerPrager( std::string const & name, Group * const parent ):
+  ElasticIsotropic( name, parent ),
+  m_defaultFrictionAngle(),
+  m_defaultDilationAngle(),
+  m_defaultCohesion(),
+  m_defaultHardening(),
+  m_friction(),
+  m_dilation(),
+  m_hardening(),
+  m_newCohesion(),
+  m_oldCohesion()
+{
+  // register default values
+
+  registerWrapper( viewKeyStruct::defaultFrictionAngleString, &m_defaultFrictionAngle )->
+    setApplyDefaultValue( 30.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Friction angle (degrees)" );
+
+  registerWrapper( viewKeyStruct::defaultDilationAngleString, &m_defaultDilationAngle )->
+    setApplyDefaultValue( 30.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Dilation angle (degrees)" );
+
+  registerWrapper( viewKeyStruct::defaultHardeningString, &m_defaultHardening )->
+    setApplyDefaultValue( 0.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Cohesion hardening/softening rate" );
+
+  registerWrapper( viewKeyStruct::defaultCohesionString, &m_defaultCohesion )->
+    setApplyDefaultValue( 0.0 )->
+    setInputFlag( InputFlags::OPTIONAL )->
+    setDescription( "Initial cohesion" );
+
+  // register fields
+
+  registerWrapper( viewKeyStruct::frictionString, &m_friction )->
+    setApplyDefaultValue( -1 )->
+    setDescription( "Yield surface slope" );
+
+  registerWrapper( viewKeyStruct::dilationString, &m_dilation )->
+    setApplyDefaultValue( -1 )->
+    setDescription( "Plastic potential slope" );
+
+  registerWrapper( viewKeyStruct::hardeningString, &m_hardening )->
+    setApplyDefaultValue( -1 )->
+    setDescription( "Hardening rate" );
+
+  registerWrapper( viewKeyStruct::newCohesionString, &m_newCohesion )->
+    setApplyDefaultValue( -1 )->
+    setPlotLevel( dataRepository::PlotLevel::LEVEL_3 )->
+    setDescription( "New cohesion state" );
+
```

Comment:
I tried to find this using VisIt, it's quire hard to fund at level3 with so many stuffs